src/main/java/frc/robot/subsystems/ball/feeder/conveyor/ConveyorImpl.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems.ball.feeder.conveyor;
+
+import java.util.concurrent.TimeUnit;
+import ca.team3161.lib.robot.LifecycleEvent;
+import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import frc.robot.RobotMap;
+
+
+public class ConveyorImpl extends RepeatingPooledSubsystem implements Conveyor {
+
+    public enum ConveyorState {
+        FEEDING,
+        OFF,
+        UNLOADING,
+        PRIMING
+    }
+
+    WPI_TalonSRX conveyorController;
+    volatile ConveyorState state;
+
+    ConveyorImpl() {
+        super(1, TimeUnit.SECONDS);
+        this.conveyorController = new WPI_TalonSRX(RobotMap.BELT_TALON_PORT);
+        this.conveyorController.setInverted(true);
+        this.state = ConveyorState.OFF;
+    }
+
+    @Override
+    public void defineResources() {
+        require(this.conveyorController);
+    }
+
+    @Override
+    public void task() {
+        if(this.state == ConveyorState.FEEDING) {
+            this.conveyorController.set(0.95d);
+        } else if(this.state == ConveyorState.UNLOADING) {
+            this.conveyorController.set(-0.95d);
+        } else if(this.state == ConveyorState.PRIMING){
+            // TODO ultrasonic sensor
+            this.conveyorController.set(-0.95d);
```

Comment:
Wouldn't PRIMING and FEEDING run the motor in the same direction?

Reply:
They will, but priming will eventually have some extra logic for the ultrasonic sensor